packages/react-sdk/src/helpers/caching/db.ts

```diff
@@ -62,59 +62,103 @@ export type GetDBInstanceOptions = {
   version?: number;
 };
 
+const getDbSchema = (db: Dexie) =>
+  JSON.stringify(
+    db.tables
+      .map(({ name, schema }) => ({
+        name,
+        schema: [
+          schema.primKey.src,
+          ...schema.indexes.map((idx) => idx.src).sort(),
+        ].join(","),
+      }))
+      .sort((a, b) => (a.name < b.name ? 1 : -1)),
+  );
+
+const hasSchemaChanged = async (db: Dexie) => {
+  const declaredSchema = getDbSchema(db);
+  const newDb = new Dexie(db.name);
+  await newDb.open();
+  const installedSchema = getDbSchema(newDb);
+  return declaredSchema !== installedSchema;
+};
+
+const checkSchema = async (version: number, schema: Record<string, string>) => {
+  const db = new Dexie("__XMTP__");
+  db.version(version).stores(schema);
+  await db.open();
+  if (await hasSchemaChanged(db)) {
+    throw new Error(
+      `The local DB schema has changed and must be upgraded.
```

Comment:
Are we only changing the schema on major version's? Seems like that is a requirement, given it requires an explicit user migration to update their schema version

Reply:
that's a good question and something i've thought about a lot. sometimes adding a new feature requires a schema change, but _adding_ to the schema doesn't usually create an issue. it's only when updating/removing an existing schema that requires a new version.

that being said, having to add/update a component prop for an SDK update could be considered a breaking change in that it does require a code change. however, it's not an involved refactor, usually just a single character change. is that a breaking change? i could be convinced that it is... or isn't.

ideally, this is something that we would manage internally so that developers don't have to think about it. how we're currently managing the schema for content type caching makes this more difficult since we're not storing all versions of the schema over time.

i'm definitely open to going down this route. it was something i originally considered but went with a simpler solution at the time.